tasks/main.yml

```diff
@@ -1,28 +1,45 @@
 ---
+# Reboot if required before installing new updates
+- name: Check if reboot is needed
+  ansible.windows.win_powershell:
+    script: |
+      $RebootPending = Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending"
+      $RebootRequired = $RebootPending `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired") `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootPending")
+      if ($RebootRequired) {
+          Write-Output "YES"
+      } else {
+          Write-Output "NO"
+      }
+  register: reboot_needed
 
-#Reboot if required before installing new updates
-- name: If reboot required before checking and installing new updates, reboot the host.
-  win_reboot:
-   reboot_timeout_sec: 1500
-  when: ansible_reboot_pending
+- name: Reboot host before check for update
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
+  when: reboot_needed.output is defined and "YES" in reboot_needed.output
 
-#Check for missing updates
-- name: Check for missing updates.
-  win_updates: state=searched
+# Check for missing updates
+- name: Check for missing updates
+  ansible.windows.win_updates:
+    category_names: "{{ update_category_names }}"
+    state: searched
   register: update_count
 
-#List the missing updates
-- name: List the updates.
-  debug: var=update_count
+# List the missing updates
+- name: List the updates
+  ansible.builtin.debug:
+    msg: "{{ update_count }}"
 
-#Install Updates
+# Install Updates
 - name: Install all Critical and security updates.
-  win_updates:
-    category_names: "{{update_category_names}}"
+  ansible.windows.win_updates:
+    category_names: "{{ update_category_names }}"
     state: installed
   register: update_result
 
+# Reboot operational system
 - name: Reboot host after installing updates.
-  win_reboot:
-    reboot_timeout_sec: 1500
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
```

Comment:
Ditto

Reply:
Considering that some reboots take a long time in the update process, I consider it more prudent to change to 30 min. The interesting side about this module is that if the reboot takes only 10 min, it will not wait 30 min, but only 10 min.